File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int n=Math.abs(x);
        int reverse=0;
        while(n>0){
            int a=n%10;
            if (reverse > (Integer.MAX_VALUE - a) / 10) {
                return 0; // Return 0 if overflow would occur
            }
            reverse=(reverse*10)+a;
            n=n/10;
        }
        return (x<0)?-reverse:reverse;
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        n=abs(x)
        reverse=0
        INT_MAX = 2**31 - 1
        while n>0:
            a=n%10
            reverse=(reverse*10)+a
            n=n//10
            if reverse>INT_MAX:
                return 0
        return reverse if x >= 0 else -reverse
        